internal/namespace/manager.go

```diff
@@ -124,3 +139,129 @@ func (nm *namespaceManager) initNamespaces(ctx context.Context, di database.Plug
 	}
 	return nil
 }
+
+func (nm *namespaceManager) validateNamespaceConfig(ctx context.Context, name string, index int, conf config.Section) error {
+	if err := core.ValidateFFNameField(ctx, name, fmt.Sprintf("namespaces.predefined[%d].name", index)); err != nil {
+		return err
+	}
+
+	if strings.ToLower(name) == "ff_system" || strings.ToLower(conf.GetString(coreconfig.NamespaceRemoteName)) == "ff_system" {
+		return i18n.NewError(ctx, coremsgs.MsgFFSystemReservedName)
+	}
+
+	mode := conf.GetString(coreconfig.NamespaceMode)
+	plugins := conf.GetStringSlice(coreconfig.NamespacePlugins)
+
+	// If the no plugins are found when querying the config, assume older config file
+	if len(plugins) == 0 {
+		// Still check to ensure at least one bc, dx, ss, and db plugin exist
+		if len(nm.bcPlugins) == 0 || len(nm.dxPlugins) == 0 || len(nm.ssPlugins) == 0 || len(nm.dbPlugins) == 0 {
+			return i18n.NewError(ctx, coremsgs.MsgNamespaceMultipartyConfiguration, name)
+		}
+		return nil
+	}
+
+	switch mode {
+	// Multiparty is the default mode when none is provided
+	case "multiparty", "":
```

Comment:
If you properly set the default, I _think_ the mode will always be set so you don't have to check for `""`